about/about-kubernetes-egress.md

```diff
@@ -0,0 +1,111 @@
+
+In the context of Kubernetes egress, NAT is used to allow pods to connect to services outside of the cluster if the pods
+have IP addresses that are not routable outside of the cluster (for example, if the pod network is an overlay).
+
+For example, if a pod in an overlay network attempts to connect to an IP address outside of the cluster, then the
+node hosting the pod uses SNAT (Source Network Address Translation) to map the non-routable source IP address of tha
+packet to the node's IP address before forwarding the packet on.  The node then maps response packets coming in the
+opposite direction back to the original pod IP address, so packets flow end-to-end in both directions, with neither
+pod or external service being aware the mapping is happening.
+
+In most clusters this NAT behavior is configured statically across the whole of the cluster. When using
+{{site.prodname}}, the NAT behavior can be configured at a more granular level for particular address ranges using [IP
+pools]({{site.baseurl}}/reference/resources/ippool). This effectively allows the scope of "non-routable" to be more
+tightly defined than just "inside the cluster vs outside the cluster", which can be useful in some enterprise deployment
+scenarios.
+
+### Egress gateways
+
+Another approach to Kubernetes egress is to route all outbound connections via one or more egress gateways. The gateways
+SNAT (Source Network Address Translation) the connections so the external service being connected to sees the connection
+as coming from the egress gateway. The main use case is to improve security, either with the egress gateway performing a
+direct security role in terms of what connections it allows, or in conjunction with perimeter firewalls (or other
+external entities). For example, so that perimeter firewalls see the connections coming from well know IP
+addresses (the egress gateways) rather than from dynamic pod IP address they don't understand.
+
+Egress gateways are not a native concept in Kubernetes itself, but are implemented by some Kubernetes network
+implementations and some service meshes. For example, Calico Enterprise provides egress gateway functionality, plus the
+ability to map namespaces (or even individual pods) to specific egress gateways. Perimeter firewalls (or other external
+security entities) can then effectively provide per namespace security controls, even though they do not have visibility
+to dynamic pod IP addresses.
+
+As an alternative approach to egress gateways, {{site.prodname}} allows you to control pod IP address ranges based on
+namespace, or node, or even at the individual pod level. Assuming no outgoing NAT is required, this provides a very
+simple way for perimeter firewalls (or other external security entities) to integrate with Kubernetes for both ingress
+and egress traffic. (Note that this approach relies on having enough address space available to sensible assign IP
```

Comment:
Note that this approach relies on having enough address space available to **sensibly** assign...